packages/orama/src/trees/flat.ts

```diff
@@ -0,0 +1,81 @@
+import { InternalDocumentID } from "../components/internal-document-id-store.js"
+import { EnumComparisonOperator, ScalarSearchableType, ScalarSearchableValue } from "../types.js"
+
+export interface FlatTree {
+  foo: string
+  numberToDocumentId: Map<ScalarSearchableValue, InternalDocumentID[]>
+}
+
+export function create(): FlatTree {
+  return {
+    foo: 'bar',
```

Comment:
Why do you need this?

Reply:
At the searching time, we don't have the `schema`. So the logic to differentiate the index is made on the "filter" type.
I mean, a tree is cast to the boolean tree if the value of the filter is a boolean. The same is true for the string. In the end, as a fallback, there's the number logic.
That should be different: we should keep the kind of tree inside the tree itself and use that value to cast to the right type.
This is a simple trick to differentiate the AVL tree from this tree, so differentiate number filters from enum ones.
Is it correct? Surely not.

Any suggestions?

Reply:
Can't you access the schema through the Orama object?